hyperspy/learn/mva.py

```diff
@@ -814,15 +814,17 @@ def _calculate_recmatrix(self, components=None, mva_type=None,):
                 mva_type, components)
 
         self._unfolded4decomposition = self.unfold()
-
-        sc = self.deepcopy()
-        sc.data = a.T.reshape(self.data.shape)
-        sc.metadata.General.title += signal_name
-        if target.mean is not None:
-            sc.data += target.mean
-        if self._unfolded4decomposition is True:
-            self.fold()
-            sc.fold()
+        try:
+            sc = self.deepcopy()
+            sc.data = a.T.reshape(self.data.shape)
+            sc.metadata.General.title += signal_name
+            if target.mean is not None:
+                sc.data += target.mean
+        finally:
+            if self._unfolded4decomposition is True:
+                self.fold()
+                sc.fold()
+            # TODO: Should self._unfolded4decomposition be set to False?
```

Comment:
The old code never set `_unfolded4decomposition` to `False` here, should it be done?